.github/actions/setup-windows-builder/action.yaml

```diff
@@ -38,8 +38,8 @@ runs:
     - name: Setup Rust toolchain
       shell: bash
       run: |
-        rustup update stable
-        rustup toolchain install stable
+        # Avoid self update to avoid CI failures: https://github.com/apache/arrow-datafusion/issues/9653
```

Comment:
should we add the same for other OS?

Reply:
@comphead I think we don't need to disable self update for other OS, the issue is window only. After the issue is solved, we can enable self update again.
@alamb